Chatto/sources/Keyboard/KeyboardUpdatesHandler.swift

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright (c) Badoo Trading Limited, 2010-present. All rights reserved.
+//
+import UIKit
+
+public protocol KeyboardInputAdjustableViewController: UIViewController {
+    var inputBarContainer: UIView { get }
+    var inputContainerBottomConstraint: NSLayoutConstraint { get }
+}
+
+public protocol KeyboardUpdatesHandlerProtocol {
+    var keyboardTrackingView: KeyboardTrackingView { get }
```

Comment:
I'd suggest to replace `keyboardTrackingView`'s type to UIView and make `KeyboardTrackingView` public